server/api/src/main/java/org/hawkular/apm/server/api/services/SpanService.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hawkular.apm.server.api.services;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.hawkular.apm.api.services.StoreException;
+import org.hawkular.apm.server.api.model.zipkin.Span;
+
+/**
+ * This interface represents the service used to store and retrieve spans.
+ *
+ * @author Pavol Loffay
+ */
+public interface SpanService {
+
+    /**
+     * This method returns span of a given id.
+     *
+     * @param tenantId The tenant
+     * @param id The id
+     * @return
+     */
+    Span getSpan(String tenantId, String id);
+
+    /**
+     * This method returns all children of a given span.
+     *
+     * @param tenantId The Tenant
+     * @param id The span
+     * @return
+     */
+    List<Span> getChildren(String tenantId, String id);
```

Comment:
Document what you return on "unexpected" outcomes. For instance, the "no index" situation, or the inconsistent List in case one or more items fail to parse, ...